cmake/modules/hunter_pkgconfig_export_target.cmake

```diff
@@ -29,15 +29,25 @@ function(hunter_pkgconfig_export_target PKG_CONFIG_MODULE)
   endif()
   add_library("${target_name}" INTERFACE IMPORTED GLOBAL)
 
+  if (BUILD_SHARED_LIBS)
```

Comment:
> I think using if(BUILD_SHARED_LIBS) should work fine here

Wait, actually it's not correct too. This module is called on the user side, and the variable should be passed from the package build phase.

`BUILD_SHARED_LIBS` variable should be used in `configure_file` here:
- https://github.com/ruslo/hunter/blob/v0.23.158/cmake/schemes/url_sha1_autogen_autotools.cmake.in#L65-L73

to generate `*Config.cmake` file:
- https://github.com/ruslo/hunter/blob/v0.23.158/scripts/pkgconfig-export-targets.cmake.in#L10

Something like:
```cmake
set(PKG_GENERATE_SHARED "${BUILD_SHARED_LIBS}")

# ...

  # Use:
  # * DEPENDS_ON_PACKAGES
  # * PKG_GENERATE_SHARED
  configure_file(
      "@HUNTER_GLOBAL_SCRIPT_DIR@/pkgconfig-export-targets.cmake.in"
      "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/cmake/${PKG_CONFIG_MODULE}/${PKG_CONFIG_MODULE}Config.cmake"
      @ONLY
  )
```

```cmake
hunter_pkgconfig_export_target("@PKG_CONFIG_MODULE@" "@PKG_GENERATE_SHARED@")
```

```cmake
function(hunter_pkgconfig_export_target PKG_CONFIG_MODULE PKG_GENERATE_SHARED)

  # ...

  if(${PKG_GENERATE_SHARED})
```

Reply:
OK, I have put these changes in and kicked off all the tests again.

Reply:
@ruslo `set(PKG_GENERATE_SHARED "${BUILD_SHARED_LIBS}")` will have to be added to 
* `url_sha1_tcl_autotools.cmake.in`
* `url_sha1_x264.cmake.in`
* `xcb.cmake.in`
* `url_sha1_autogen_autotools.cmake.in`
* `url_sha1_autotools.cmake.in`

Is that correct, or should I just use `BUILD_SHARED_LIBS` in `pkgconfig-export-targets.cmake.in` directly?

Reply:
> should I just use BUILD_SHARED_LIBS in pkgconfig-export-targets.cmake.in directly?

Yes, `BUILD_SHARED_LIBS` can be used here. I just prefer an explicitly new name, to avoid confusion. Five files is not a big list. Up to you, I'm okay with both choices.

Reply:
I have added it in the 5 places